examples/sugarscape_g1mt/sugarscape_g1mt/agents.py

```diff
@@ -340,3 +340,9 @@ def calculate_MRS(self, sugar=None, spice=None):
             spice = self.spice
         # See GAS page 102 equation 3.
         return (spice / self.metabolism_spice) / (sugar / self.metabolism_sugar)
+
+    def step(self):
```

Comment:
This changes the simulation implementation complete. Each agent now does all the following steps before the next again. This may cause first-move-advantage problem.

Reply:
Yes, I plan to get back to each part of the process fix but first had to figure out why the time wasn't advancing, it has to call through `schedule.step` if they want the`mesa.time` code to be implemented.

Reply:
A challenge is the model is really set up as stagedactivationbytype...which we do not have.

Reply:
We can just manually shuffle the agents, and advance the time inside the model step. Simplest solution for now.